src/callbacks_step/analysis_surface_integral_2d.jl

```diff
@@ -0,0 +1,187 @@
+    end
+
+    function AnalysisSurfaceIntegral(semi, boundary_symbols::Vector{Symbol}, variable)
+        @unpack boundary_symbol_indices = semi.boundary_conditions
+        indices = Vector{Int}()
+        for name in boundary_symbols
+            append!(indices, boundary_symbol_indices[name])
+        end
+        sort!(indices)
+
+        return new{typeof(semi), typeof(variable)}(semi, indices,
+                                                   variable)
+    end
+end
+
+struct ForceState{RealT <: Real}
+    psi::Tuple{RealT, RealT} # Unit vector normal or parallel to freestream
+    rhoinf::RealT
+    uinf::RealT
+    linf::RealT
+end
+
+struct LiftCoefficientPressure{RealT <: Real}
+    force_state::ForceState{RealT}
+end
+
+struct DragCoefficientPressure{RealT <: Real}
+    force_state::ForceState{RealT}
+end
+
+""" 
+    LiftCoefficientPressure(aoa, rhoinf, uinf, linf)
+
+Compute the lift coefficient 
+```math
+C_{L,p} \coloneqq \frac{\oint_{\partial \Omega} p \boldsymbol n \cdot \psi_L \, \mathrm{d} S}
+                        {0.5 \cdot \rho_{\infty} \cdot U_{\infty}^2 \cdot L_{\infty}}
+```
+based on the pressure distribution along a boundary.
+Supposed to be used in conjunction with [`AnalysisSurfaceIntegral`](@ref) 
+which stores the boundary information and semidiscretization.
+
+- `aoa::Real`: Angle of attack in radians (for airfoils etc.)
+- `rhoinf::Real`: Free-stream density
+- `uinf::Real`: Free-stream velocity
+- `linf::Real`: Reference length of geometry (e.g. airfoil chord length)
+"""
+function LiftCoefficientPressure(aoa, rhoinf, uinf, linf)
+    # psi_lift is the normal unit vector to the freestream direction.
+    # Note: The choice of the normal vector psi_lift = (-sin(aoa), cos(aoa))
+    # leads to positive lift coefficients for positive angles of attack for airfoils.
+    # Note that one could also use psi_lift = (sin(aoa), -cos(aoa)) which results in the same 
+    # value, but with the opposite sign.
+    psi_lift = (-sin(aoa), cos(aoa))
+    return LiftCoefficientPressure(ForceState(psi_lift, rhoinf, uinf, linf))
+end
+
+""" 
+    DragCoefficientPressure(aoa, rhoinf, uinf, linf)
+
+Compute the drag coefficient 
+```math
+C_{D,p} \coloneqq \frac{\oint_{\partial \Omega} p \boldsymbol n \cdot \psi_D \, \mathrm{d} S}
+                        {0.5 \cdot \rho_{\infty} \cdot U_{\infty}^2 \cdot L_{\infty}}
+```
+based on the pressure distribution along a boundary.
+Supposed to be used in conjunction with [`AnalysisSurfaceIntegral`](@ref) 
+which stores the boundary information and semidiscretization.
+
+- `aoa::Real`: Angle of attack in radians (for airfoils etc.)
+- `rhoinf::Real`: Free-stream density
+- `uinf::Real`: Free-stream velocity
+- `linf::Real`: Reference length of geometry (e.g. airfoil chord length)
+"""
+function DragCoefficientPressure(aoa, rhoinf, uinf, linf)
+    # `psi_drag` is the unit vector in direction of the freestream.
+    psi_drag = (cos(aoa), sin(aoa))
```

Comment:
so it is the tangent vector, yes? As opposed to the normal vector used in the lift computation. Might be worth mentioning that the same notation `psi` is used for two purposes.